src/static.jl

```diff
@@ -260,25 +260,27 @@ Base.any(::Tuple{Vararg{True}}) = true
 Base.any(::Tuple{Vararg{Union{True,False}}}) = true
 Base.any(::Tuple{Vararg{False}}) = false
 
-Base.@pure nstatic(::Val{N}) where {N} = ntuple(i -> StaticInt(i), Val(N))
+@inline nstatic(::Val{N}) where {N} = ntuple(i -> StaticInt(i), Val(N))
```

Comment:
Why not
```julia
@inline nstatic(::Val{N}) where {N} = ntuple(StaticInt, Val(N))
```
Removing an anonymous function probably makes things ever so slightly easier on the compiler. Slightly fewer allocations (actual times are very noisy):
```julia
> julia -O3 --startup=no -e "@time begin; foo(::Val{N}) where {N} = ntuple(i -> Val(i), Val{N}()); foo(Val(8)); end"                                                                 (base)
  0.013163 seconds (14.18 k allocations: 1.021 MiB, 114.01% compilation time)
> julia -O3 --startup=no -e "@time begin; foo(::Val{N}) where {N} = ntuple(Val, Val{N}()); foo(Val(8)); end"                                                                         (base)
  0.013611 seconds (11.99 k allocations: 836.604 KiB, 113.98% compilation time)
```

Reply:
Dang! That is just an obvious improvement on what I was doing.